public-api/src/main/java/org/pmiops/workbench/publicapi/DataBrowserController.java

```diff
@@ -557,26 +567,41 @@ public SurveyQuestionResult apply(AchillesResult o) {
             if (criteriaList.size() >= 1) {
                 criteriaList.remove(parent);
             }
+            if (parent == null) {
+                throw new DataNotFoundException("Cannot find rolled up counts of this concept");
+            }
             response.setParent(TO_CLIENT_CBCRITERIA.apply(parent));
             Multimap<Long, CBCriteria> parentCriteria = Multimaps
                     .index(criteriaList, CBCriteria::getParentId);
             return ResponseEntity.ok(response);
+        } else {
+            throw new DataNotFoundException("Cannot find rolled up counts of this concept");
         }
-        return ResponseEntity.ok(response);
     }
 
     @Override
     public ResponseEntity<CriteriaListResponse> getCriteriaChildren(Long parentId) {
-        CdrVersionContext.setCdrVersionNoCheckAuthDomain(defaultCdrVersionProvider.get());
+        try {
+            CdrVersionContext.setCdrVersionNoCheckAuthDomain(defaultCdrVersionProvider.get());
+        } catch(NullPointerException ie) {
+            throw new DataNotFoundException("Cdr Version table is not available. Please check if the database is up and version is right.");
+        }
         List<CBCriteria> criteriaList = criteriaDao.findCriteriaChildren(parentId);
         CriteriaListResponse criteriaListResponse = new CriteriaListResponse();
+        if (criteriaList == null) {
```

Comment:
Not Possible. criteriaDao will return an empty list of nothing is found.

Reply:
Changed. Thanks.